woocommerce/woocommerce
# Tweak CSS to allow wc_input_price class to work in the settings context

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This adds CSS to allow the `wc_input_price` class to work correctly in the settings page. We have a demand in an extension to add a price field to the settings page.

Part of https://github.com/woocommerce/woocommerce-min-max-quantities/issues/380



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to WooCommerce > Settings > General and define `,` as the decimal separator. 
2. Install the following snippet that adds a custom price field in the WooCommerce > Settings > Products tab: 

```
add_filter( 'woocommerce_products_general_settings', 'add_settings', 60 );

function add_settings( $settings ) {

	$settings_to_add = array(
		array(
					'title' => __( 'Test section', 'woocommerce' ),
					'type'  => 'title',
					'desc'  => '',
					'id'    => 'product_test_section',
				),
		array(
			'name'     => __( 'Custom Price', 'woocommerce-min-max-quantities' ),
			'desc'     => __( 'A Custom Price field that does nothing.', 'woocommerce-min-max-quantities' ),
			'id'       => 'woocommerce_custom_price_test_field',
			'type'     => 'text',
			'class'    => 'wc_input_price',
			'desc_tip' => true,
		),
		array(
					'type' => 'sectionend',
					'id'   => 'product_inventory_options',
				),
	);
	$new_settings = array_merge( $settings, $settings_to_add );

	return $new_settings;
}
```
3. Go to WooCommerce > Settings > Products and fill in a price with an invalid decimal separator, like `10.5`. 
4. Notice that the error is not aligned to the field. 
5. Pull and build this branch. 
6. Notice that the error is correctly aligned now. 



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
